README.md

```diff
@@ -215,15 +215,53 @@ drop messages or crash, also based on configuration.
 Subscriptions are managed using `eredis_sub:subscribe/2` and
 `eredis_sub:unsubscribe/2`. When Redis acknowledges the change in
 subscription, a message is sent to the controlling process for each
-channel.
+channel. Then, eredis sends a message on the form `{subscribed, Channel,
+ClientPid}` to the controlling process, which must be acked using
+`eredis_sub:ack_message/1`.
 
 eredis also supports Pattern Subscribe using `eredis_sub:psubscribe/2`
-and `eredis_sub:unsubscribe/2`. As with normal subscriptions, a message
-is sent to the controlling process for each channel.
+and `eredis_sub:punsubscribe/2`. As with normal subscriptions, a message
+is sent to the controlling process for each channel and eredis sends a
+message on the form `{subscribed, Channel, ClientPid}` to the
+controlling process, which must be acked using `eredis_sub:ack_message/1`.
+
+The controlling process is also notified in case of reconnection attempts or
+failures. See `test/eredis_pubsub_SUITE.erl` for examples.
+
+Starting with version 1.5, eredis automatically resubscribes after reconnect.
+Then, the controlling process will receive `{subscribed, Channel, ClientPid}`
+messages again for every channel and pattern subscribed to. These must also be
+acked.
+
+Here is a list of all the messages that are sent to the controlling process.
+Some messages don't need to be acked, but it does not harm doing so.
```

Comment:
I wonder about this, but maybe I just don't fully understand.

Since ack_message/1 only takes the eredis_sub_client Pid as argument, I guess it doesn't specify which/what message it is acknowledging. If the eredis_sub side does not expect a message to be acknowledged, but gets an ack, won't that ack ever be wrongfully assumed to be towards a different message?

I guess if a strict queue is used maybe not?

Reply:
You're right, it's just a queue. When a message is acked, the client sends the next message. If the queue is empty, an ack does nothing.

Reply:
@e-kmisk Do you think this PR is OK and ready to merge?

Reply:
Now it looks ok, I don't see any issues with it.

We will do another test with the connection errors today or Monday, patching the BEAM files in our image with these new ones.

Reply:
Still working on testing it, we are having some unrelated issues stopping me from trying.
Hopefully I can get to it today.

Reply:
It's covered by unit tests, so I think it's safe to merge this one. WDYT?

Reply:
No hurry though. I can wait some more days.